lib/rack/multipart.rb

```diff
@@ -17,7 +17,7 @@ module Multipart
     BROKEN_QUOTED = /^#{CONDISP}.*;\sfilename="(.*?)"(?:\s*$|\s*;\s*#{TOKEN}=)/i
     BROKEN_UNQUOTED = /^#{CONDISP}.*;\sfilename=(#{TOKEN})/i
     MULTIPART_CONTENT_TYPE = /Content-Type: (.*)#{EOL}/ni
-    MULTIPART_CONTENT_DISPOSITION = /Content-Disposition:.*\s+name="?([^\";]*)"?/ni
+    MULTIPART_CONTENT_DISPOSITION = /Content-Disposition:.*\s+name=(#{TOKEN}|"[^\";]*")/ni
```

Comment:
it might be more correct to match separately with: `(?:(#{TOKEN})|"([^\";]*)")`
and then not do a gsub!
```
match = head.match(MULTIPART_CONTENT_DISPOSITION)
match[1] || match[2] || head[MULTIPART_CONTENT_ID, 1]
```